### PR TITLE
fix(report): validate and stabilize CycloneDX exports

### LINKED ISSUE #1130: [Feature]: Validate CycloneDX exports against the official schema
## Product outcome

Make every Lopper CycloneDX artifact consumable by standards-conformant tooling and prevent invalid SBOM changes from reaching a release.

## Opportunity type

Existing-feature hardening and interoperability.

## Evidence

- The v1.8 formatter declares CycloneDX 1.6 and emits the official schema URL, but [PR #1106](https://github.com/ben-ranford/lopper/pull/1106) explicitly deferred schema validation.
- CycloneDX states that JSON Schema is the conformance mechanism for JSON BOMs: [official schema guidance](https://cyclonedx.org/guides/sbom/external-references).
- The formatter is centralized in `internal/report/format_cyclonedx.go`, and deterministic golden tests already provide a narrow integration seam.

## Scope

Pin the official CycloneDX 1.6 JSON schema for CI/test use, validate representative and edge-case exports, and make schema validation part of the release gate. Keep runtime export self-contained; validation belongs in tests and an optional diagnostic command, not an implicit network call.

## Acceptance criteria

- [ ] CI validates representative empty, single-repository, policy, baseline, vulnerability, and duplicate-name BOM fixtures against the pinned official schema.
- [ ] The schema artifact is checksum-pinned or supplied by a version-pinned validator.
- [ ] `$schema`, `bomFormat`, `specVersion`, component references, licenses, properties, and timestamps conform.
- [ ] A developer-facing validation command or test target reports actionable JSON paths on failure.
- [ ] Export ordering remains deterministic across repeated runs.
- [ ] CI and runtime validation perform no network access.

## Non-goals

Adding dependency versions/PURLs, SPDX output, portfolio aggregation, or signed attestations; those require separate model and release work.

## Dependencies

Coordinate with [#1095](https://github.com/ben-ranford/lopper/issues/1095) and the preview-graduation issue.

### LINKED ISSUE #1153: [Bug]: Keep duplicate CycloneDX components deterministic and correctly attributed
## Summary

CycloneDX generation loses identity when a report contains multiple dependency rows with the same `language` and `name`:

1. Baseline deltas are indexed only by `language+name`, so both components receive the last duplicate's delta.
2. Duplicate rows have no total sort order, so input permutation changes component order and which row receives the unsuffixed `bom-ref`.

Both failures reproduced twice. Official CycloneDX 1.6 schema validation passed for all tested fixtures, so this is an attribution/determinism defect distinct from #1130's schema gate.

## Reproduction steps

Construct a report with two same-key dependencies carrying distinct metrics/imports:

```go
report.Report{
    Dependencies: []report.DependencyReport{
        {Language: "js-ts", Name: "dup", UsedExportsCount: 1, TotalExportsCount: 2, UsedImports: []report.ImportUse{{Name: "a", Module: "m/a"}}},
        {Language: "js-ts", Name: "dup", UsedExportsCount: 2, TotalExportsCount: 3, UsedImports: []report.ImportUse{{Name: "b", Module: "m/b"}}},
    },
    BaselineComparison: &report.BaselineComparison{Dependencies: []report.DependencyDelta{
        {Language: "js-ts", Name: "dup", UsedExportsCountDelta: 1},
        {Language: "js-ts", Name: "dup", UsedExportsCountDelta: 9},
    }},
}
```

Format it as CycloneDX, inspect each component's `lopper:baseline:used-exports-count-delta`, then repeat with the dependency input order reversed and compare the rendered artifacts.

## Expected behavior

Each duplicate component keeps its corresponding metrics/delta, and logically identical report sets produce byte-for-byte deterministic output regardless of input order.

## Actual behavior

Both components receive delta `9` (`component0=9 component1=9`). Reversing the input changes the leading component's `lopper:used-imports` value and `bom-ref` suffix assignment.

## Logs and output

Each repro passed the gate twice with these stable signals:

```text
duplicate baseline delta collapsed onto both duplicate components
component0=9 component1=9
non-deterministic duplicate component ordering
```

Schema-invalid output was not observed in empty, policy, baseline, vulnerability, license, timestamp, or duplicate-name controls.

## Version / commit

`a9effdbadd3b570f1e8ed05ec35e282f1efb56ea`

## Environment

Darwin 25.5.0 arm64; Go 1.26.5; offline CycloneDX 1.6 validator.

## Affected areas

- `internal/report/format_cyclonedx.go`
- `internal/report/format_csv.go` sorting helper
- CycloneDX formatter regression tests

## Acceptance criteria

- [ ] Duplicate dependency rows receive a deterministic total order based on stable row content.
- [ ] Baseline deltas are matched to the correct duplicate instance rather than collapsed by bare `language+name`.
- [ ] `bom-ref` values remain unique and stable across input permutations.
- [ ] Repeated formatting of equivalent report sets is byte-for-byte deterministic.
- [ ] CycloneDX 1.6 schema validation and existing unique-name behavior remain green.

### PR DESCRIPTION
**Pull request by ben-ranford**
## Summary

Harden the preview CycloneDX 1.6 exporter with a fully offline official-schema release gate and deterministic handling for duplicate dependency instances. This keeps standards regressions out of releases while preserving the existing unique-name ordering contract.

Closes #1130
Closes #1153

## Changes

- Pin the official CycloneDX 1.6 root, SPDX, and JSF schemas at upstream tag `1.6` / commit `55343ba19dee1785acf1ce9191540d5fd7b590db`, with SHA-256 and Git blob provenance in test metadata.
- Validate empty, single, policy, baseline, vulnerability, license, timestamp, and duplicate fixtures with network access blocked; surface failures with JSON paths.
- Add the discoverable `make cyclonedx-schema-check` target and repeated-run byte-determinism assertions.
- Canonically order duplicate same-key components and baseline deltas so each instance receives one stable delta and stable unique `bom-ref` suffix.
- Keep `LOP-FEAT-0013` in preview/default-off state.

## Validation

Commands and checks run:

```bash
make ci
make demos-check
make cyclonedx-schema-check
GOTOOLCHAIN=go1.26.5 go test ./internal/report -run 'TestFormatCycloneDXJSONKeepsDuplicate|TestFormatCycloneDXJSONDeterministic' -count=10
```

Additional manual validation:

- The pre-commit hook reran the full `make ci` suite over the staged snapshot and passed.
- Official schema checks completed with HTTP disabled and verified all three pinned SHA-256 digests.
- `act pull_request -W .github/workflows/ci.yml --job verify` was not run; the complete native gate passed and remote PR checks will exercise the workflow.

## Risk and compatibility

- Breaking changes: None. Existing language/name ordering remains the primary order; stable content only breaks duplicate ties.
- Migration required: None.
- Performance impact: Limited to CycloneDX formatting when sorting component and delta content.
- Memory benchmark impact: Passed; measured report benchmark bytes/op and allocs/op remained effectively unchanged.

## Checklist

- [x] Tests added/updated for behavior changes
- [x] Docs updated (README/docs/schema) if needed
- [x] `memory-approved` requested/applied if intentional memory benchmark regressions exceed CI thresholds
- [x] No unrelated changes included
- [x] Ready for review

### CODE REVIEW

File: internal/report/format_cyclonedx.go [hunk 1]
```diff
@@ -381,18 +389,71 @@ func sortedCycloneDXProperties(props []cycloneDXProperty) []cycloneDXProperty {
 	})
 }
 
-func cycloneDXBaselineDeltaByDependency(reportData Report) map[string]DependencyDelta {
+func sortedCycloneDXDependencies(dependencies []DependencyReport) []DependencyReport {
+	return sortedCycloneDXCopy(dependencies, func(left, right DependencyReport) bool {
+		return cycloneDXDependencySortKey(left) < cycloneDXDependencySortKey(right)
+	})
+}
```
[the remark this answers is not in the document]
Comment (ben-ranford) [reply internal/report/format_cyclonedx.go:417]: fixed in 761d8a9

File: internal/report/format_cyclonedx.go [hunk 2]
```diff
@@ -381,18 +389,71 @@ func sortedCycloneDXProperties(props []cycloneDXProperty) []cycloneDXProperty {
...
 		dep := DependencyReport{Name: delta.Name, Language: delta.Language}
-		deltas[dependencyKey(dep)] = delta
+		key := dependencyKey(dep)
+		deltas[key] = append(deltas[key], delta)
+	}
+	for key, values := range deltas {
+		deltas[key] = sortedCycloneDXCopy(values, func(left, right DependencyDelta) bool {
+			return cycloneDXBaselineDeltaSortKey(left) < cycloneDXBaselineDeltaSortKey(right)
+		})
 	}
```
[the remark this answers is not in the document]
Comment (ben-ranford) [reply internal/report/format_cyclonedx.go:450]: fixed in 761d8a9

File: internal/report/format_cyclonedx_test.go [hunk 3]
```diff
@@ -365,3 +365,60 @@ func TestAppendCycloneDXJSONPropertySkipsMarshalErrors(t *testing.T) {
...
+		{Kind: DependencyDeltaChanged, Language: "js-ts", Name: "dup", UsedExportsCountDelta: 1},
+		{Kind: DependencyDeltaChanged, Language: "js-ts", Name: "dup", UsedExportsCountDelta: 9},
+	}
+
+	permutations := []Report{
+		{Dependencies: dependencies, BaselineComparison: &BaselineComparison{Dependencies: deltas}},
+		{Dependencies: []DependencyReport{dependencies[1], dependencies[0]}, BaselineComparison: &BaselineComparison{Dependencies: deltas}},
+		{Dependencies: dependencies, BaselineComparison: &BaselineComparison{Dependencies: []DependencyDelta{deltas[1], deltas[0]}}},
+		{Dependencies: []DependencyReport{dependencies[1], dependencies[0]}, BaselineComparison: &BaselineComparison{Dependencies: []DependencyDelta{deltas[1], deltas[0]}}},
+	}
```
[the remark this answers is not in the document]
Comment (ben-ranford) [reply internal/report/format_cyclonedx_test.go]: fixed in 761d8a9
